reconcile/openshift_base.py

```diff
@@ -239,12 +239,17 @@ def apply(dry_run, oc_map, cluster, namespace, resource_type, resource,
         except FieldIsImmutableError:
             # Add more resources types to the list when you're
             # sure they're safe.
-            if resource_type not in ['Route', 'Service']:
+            if resource_type not in ['Route', 'Service', 'StatefulSet']:
                 raise
 
+            cascade = resource_type not in ['StatefulSet']
```

Comment:
This is begging for a mapping...
```python
supported_resource_types = {
    'Route': {
        'cascade': False
    },
    'Service': {
        'cascade': False
    },
    'StatefulSet': {
        'cascade': True
    },
}
```
No need to maintain 2 lists of resources_types

Reply:
very nice. implemented.